### PR TITLE
Visualize check view with check plot

### LINKED ISSUE #14537: Visualize check thresholds with giraffe threshold viewer
Checks should be visualized with CheckPlot.

### LINKED ISSUE #14586: Disable view type dropdown when in check view.
We will only be supporting line type visualization for the check view for now, so in alerting view mode, view type dropdown should be disabled.

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #14586 
Closes #14537

### CODE REVIEW

File: ui/src/shared/components/ThresholdMarkers.tsx [hunk 1]
```diff
@@ -50,41 +62,40 @@ const ThresholdMarkers: FunctionComponent<Props> = ({
...
+                key={index}
+                yScale={yScale}
+                yDomain={yDomain}
+                threshold={threshold}
+                onChangeMinPos={onChangeMinPos}
+                onChangeMaxPos={onChangeMaxPos}
+              />
+            )
+          default:
+            return null
```
Comment (121watts) [inline ui/src/shared/components/ThresholdMarkers.tsx]: The user should never get here, right?  Perhaps throw an error here for bug tracking / development purposes?

File: ui/src/shared/components/ViewSwitcher.tsx [hunk 2]
```diff
@@ -155,6 +156,19 @@ const ViewSwitcher: FunctionComponent<Props> = ({
...
+          loading={loading}
+          timeZone={timeZone}
+          viewProperties={properties}
+        >
+          {config => <Plot config={config} />}
+        </CheckPlot>
+      )
+
     default:
       return <div />
```
Comment (121watts) [inline ui/src/shared/components/ViewSwitcher.tsx]: Same here

File: http/swagger.yml [hunk 3]
```diff
@@ -8954,19 +8959,54 @@ components:
               type: boolean
             level:
               $ref: "#/components/schemas/CheckStatusLevel"
-    Threshold:
+    ThresholdBase:
       properties:
         level:
          $ref: "#/components/schemas/CheckStatusLevel"
         allValues:
```
Comment (chnn) [inline http/swagger.yml]: From our discussion earlier, sounds like this field doesn't exist anymore.

File: ui/src/shared/components/CheckPlot.tsx [hunk 4]
```diff
@@ -132,4 +143,26 @@ const CheckPlot: FunctionComponent<Props> = ({
...
-export default CheckPlot
+const mstp = (state: AppState): StateProps => {
+  const {
+    checks: {
+      current: {check},
+    },
+  } = state
+
+  let thresholds = []
+  if (check.type == 'threshold') {
```
Comment (chnn) [inline ui/src/shared/components/CheckPlot.tsx]:
```suggestion
  if (check.type === 'threshold') {
```

File: ui/src/shared/components/ThresholdMarkers.tsx (hunk 1 above)
Comment (chnn) [reply ui/src/shared/components/ThresholdMarkers.tsx]: make it a compile time error!

```
      const neverThreshold: never = threshold

      throw new Error(`unexpected threshold type "${(neverThreshold as any).type}"`)
```
luv 2 typescript

File: http/swagger.yml (hunk 3 above)
Comment (ebb-tide) [reply http/swagger.yml]: I haven't got to those changes yet.

**Pull request merged**